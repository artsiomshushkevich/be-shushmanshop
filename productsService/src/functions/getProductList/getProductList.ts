import type { ValidatedEventAPIGatewayProxyEvent } from '@libs/api-gateway';
import { formatJSONResponse } from '@libs/api-gateway';
import { productsModel } from '@models/products';

export const getProductList: ValidatedEventAPIGatewayProxyEvent<null> = async () => {
    const products = await productsModel.getList();

    return formatJSONResponse(products);
};
