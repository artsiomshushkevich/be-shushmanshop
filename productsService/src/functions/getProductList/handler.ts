import { middyfy } from '@libs/lambda';
import { getProductList } from './getProductList';

export const main = middyfy(getProductList);
