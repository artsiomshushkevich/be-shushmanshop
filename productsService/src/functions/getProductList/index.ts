import { handlerPath } from '@libs/handler-resolver';
import { LamdaConfig } from '@libs/types';

export const getProductList: LamdaConfig = {
    handler: `${handlerPath(__dirname)}/handler.main`,
    events: [
        {
            http: {
                method: 'get',
                path: 'product/available',
                cors: true
            }
        }
    ]
};
