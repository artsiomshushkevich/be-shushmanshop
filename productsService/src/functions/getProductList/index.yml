getProductList:
    handler: src/functions/getProductList/handler.main
    events:
        - http:
              method: GET
              path: /product/available
