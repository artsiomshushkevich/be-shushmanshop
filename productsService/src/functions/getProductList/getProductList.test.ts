import { getProductList } from './getProductList';
import event from './mock.json';

describe('getProductList', () => {
    test('Should return correct response', async () => {
        // @ts-ignore
        const response = await getProductList(event);

        expect(response).toMatchSnapshot();
    });
});
