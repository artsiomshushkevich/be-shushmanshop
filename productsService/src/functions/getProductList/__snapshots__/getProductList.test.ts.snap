// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getProductList Should return correct response 1`] = `
{
  "body": "[{"id":"1","title":"Audi","description":"E-tron 2021","price":30000},{"id":"2","title":"Volkswagen","description":"Golf E","price":25000},{"id":"3","title":"BYD","description":"Excellence","price":45000},{"id":"4","title":"Tesla","description":"Model S 2022","price":57000}]",
  "statusCode": 200,
}
`;
